src/autolabel/labeler.py

```diff
@@ -53,15 +57,28 @@ class LabelingAgent:
     def __init__(
         self,
         config: Union[AutolabelConfig, str, dict],
-        cache: Optional[bool] = True,
+        cache: Optional[bool] = False,
         example_selector: Optional[BaseExampleSelector] = None,
         create_task: Optional[bool] = True,
         console_output: Optional[bool] = True,
+        generation_cache: Optional[BaseCache] = None,
+        transform_cache: Optional[BaseCache] = None,
     ) -> None:
         self.create_task = create_task
         self.db = StateManager() if self.create_task else None
-        self.generation_cache = SQLAlchemyGenerationCache() if cache else None
-        self.transform_cache = SQLAlchemyTransformCache() if cache else None
+        self.generation_cache = generation_cache
+        self.transform_cache = transform_cache
+        if cache:
+            logger.warning(
+                f"cache parameter is deprecated and will be removed soon. Please use generation_cache and transform_cache instead."
+            )
+            self.generation_cache = (
+                generation_cache if generation_cache else SQLAlchemyGenerationCache()
```

Comment:
What would be the way to disable caches if cache parameter is going to be depracated?

Reply:
The user can pass the cache objects directly to generation_cache and transform_cache to use the cache if needed.